logins-api/android/LoginsApi/loginsapi/src/test/java/org/mozilla/sync15/logins/MemoryLoginsTest.kt

```diff
@@ -0,0 +1,199 @@
+/* Copyright 2018 Mozilla
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License. */
+package org.mozilla.sync15.logins
+
+import org.junit.Assert
+import org.junit.Test
+import org.junit.Assert.*
+import java.util.*
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.Semaphore
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicReference
+
+class MemoryLoginsTest {
+
+    fun getTestStore(): LoginsStorage {
+        return MemoryLoginsStorage(listOf(
```

Comment:
How hard is it to parameterize this test by the underlying storage factory?  That's what we really want, right -- to verify that the in-memory and persistent stores do the same thing.  It's not critical to do this for this PR, but I want us to be driving toward this behaviour.

Reply:
I think you could subclass the test and have `getTestStore()` return a different type, and have everything work. That was my intention when writing this, although I agree that at a minimum needs a comment, and probably isn't great even then. This is part of why I asked about the existence of an `add` method, since as it is it's not clear how we'd get a MentatLoginsStore with the desired data in it short of some process to copy one from a fixtures directory or something, which seems more fragile (since it's less obviously the same data as the in-memory store).

Additionally, something about lock/unlock would have to be figured out.